cdap-client/src/main/java/co/cask/cdap/client/MetricsClient.java

```diff
@@ -133,6 +133,23 @@ public MetricQueryResult query(List<String> metrics, List<String> groupBys, Map<
    * @throws IOException if a network error occurred
    * @throws UnauthorizedException if the request is not authorized successfully in the gateway server
    */
+  public MetricQueryResult query(List<String> metrics, List<String> groupBys, Map<String, String> tags,
```

Comment:
Also, it seems weird to always require user to pass in a list of metrics (even though the user can simply do `ImmutableList.of(theirMetric)`.
I think its quite common to simply query for a single metric.